src/Services/Processor/Substrate/Codec/Decoder.php

```diff
@@ -181,15 +181,29 @@ public function collectionStorageKey(string $data): array
         ];
     }
 
+    public function getValue(array $data, array $keys, ?string $prefix = null, ?string $suffix = null): mixed
+    {
+        foreach ($keys as $key) {
+            $keyValue = $prefix ? "{$prefix}.{$key}" : $key;
```

Comment:
It be simplified like this 
```suggestion
            $keyValue = "{$prefix}.{$key}.{$suffix}";
```

Example
null.'test'.null = `test`

Reply:
The way you did it broke, note the `.` are inside "" in the suggestions

Reply:
Didn't know that it will break, i just did test on tinker and assume that this case would work for real code as well. What exactly broke?

Reply:
You are assuming that `.` is joining the strings, but the `.` is part of the string....
"null.test.null" will give ".test." which is invalid it should be just "test"

Reply:
Note that your code is:
"{$prefix}.{$key}.{$suffix}" = "null.test.null"
And not null.'test'.null